src/utils/swapErrorToUserReadableMessage.tsx

```diff
@@ -22,6 +22,19 @@ export function swapErrorToUserReadableMessage(error: any): string {
     error = error.error ?? error.data?.originalError
   }
 
+  // The 4001 error code doesn't capture the case where users reject a transaction for all wallets,
+  // so we need to parse the reason for these special cases:
+  if (
+    // For rainbow wallet:
+    (reason?.match(/request/i) && reason?.match(/reject/i)) ||
+    // For frame wallet: https://frame.sh/
+    reason?.match(/declined/i) ||
+    // For safepal:
+    reason?.match('cancelled by user')
```

Comment:
use `/cancelled by user/i` instead of a string so it's clear that it is case-insensitive.